### ISSUE TITLE
Linting warnings are not a fun developer experience.

### ISSUE THREAD

**Issue by wraithgar** [ipfs/community#306]
Currently in js-ipfs there are 98 linting warnings in master. I don't know when the decision was made not to fail CI on these but I think it's worth revisiting this.

As a developer wanting to contribute to a repo if I have to scroll through several pages of output to try to see if any of my changes introduced errors is very frustrating.  This is compounded by the fact that linting fails locally on warnings but passes CI.

I think we should fix the warnings in all the repos and enforce them in CI.

**Comment by VictorBjelkholm** [comment]
Absolutely, I agree. Something quick we can do now that wouldn't affect things as far as I know, is to just not display warnings at all. 

The alternative solution is to turn warnings into errors, but then suddenly linting on CI is gonna start breaking everywhere, even for unrelated changes and that feels a bit too much.
